### PR TITLE
Create login hook with test login page

### PR DESCRIPTION
**Pull request by Heshoo-ksh**
Note: Files [TestUseLogin.tsx, Page.tsx] are provided for testing purposes only.
To test, navigate to: http://localhost:3000/testLogin. A correct username and password will be pre-populated, and you can change them as needed to test the API. You can also adjust these values directly in the code. Upon successful login, the page will display the full user record, including the ID, for further testing in Postman or MongoDB if the reviewer wishes.

Important: Please only review User.ts and useLogin.ts.

### CODE REVIEW

File: src/interfaces/User.ts
Comment (malabudi) [inline src/interfaces/User.ts:1]: nice idea with the interface!

File: src/hooks/TestUseLogin.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  useEffect(() => {}, [login]);
+
+  return (
+    <div>
+      <h2>Test Login Hook</h2>
+      <div>
+        <label>Email:</label>
+        <input
+          type="text"
```
Comment (malabudi) [inline src/hooks/TestUseLogin.tsx:24]: no change needed here and let moe this but for type there is a type="email" in HTML which comes with email validation ready out of the box

**Review by malabudi — APPROVED** [review]
just be sure to delete the test files before you merge, nice job!

File: src/hooks/TestUseLogin.tsx (hunk 1 above)
Comment (Heshoo-ksh) [reply src/hooks/TestUseLogin.tsx]: Nice!!! thats def will help Moe out a  bunch, I let him know

**Pull request merged**